fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(userId);
+  }, [userId]);
```

Comment:
This doesn't work. It can lead to infinite re-renders, as we learned in the typescript-closures exercise. You have to list `getUser` as a dependency and then wrap `getUser` in a `useCallback`. Or, you can just move `getUser` up inside the effect function (which is usually what people do for this).

Did you get warnings on this? You should have. If not, we need to fix your VS Code. Let me know so I can help.